src/Raven.Client/Documents/Indexes/AdditionalAssembly.cs

```diff
@@ -0,0 +1,178 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="AbstractIndexCreationTask.cs" company="Hibernating Rhinos LTD">
+//     Copyright (c) Hibernating Rhinos LTD. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Client.Documents.Indexes
+{
+    public class AdditionalAssembly : IDynamicJson
+    {
+        public string AssemblyName { get; private set; }
+
+        public string AssemblyPath { get; private set; }
+
+        public string PackageName { get; private set; }
+
+        public string PackageVersion { get; private set; }
+
+        public string PackageSourceUrl { get; private set; }
+
+        public HashSet<string> Usings { get; private set; }
+
+        private AdditionalAssembly()
+        {
+        }
+
+        public override bool Equals(object obj)
+        {
+            return obj is AdditionalAssembly assembly &&
+                   AssemblyName == assembly.AssemblyName &&
+                   AssemblyPath == assembly.AssemblyPath &&
+                   PackageName == assembly.PackageName &&
+                   PackageVersion == assembly.PackageVersion &&
+                   PackageSourceUrl == assembly.PackageSourceUrl &&
+                   EqualityComparer<HashSet<string>>.Default.Equals(Usings, assembly.Usings);
+        }
+
+        public override int GetHashCode()
+        {
+            int hashCode = -585367404;
+            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(AssemblyName);
+            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(AssemblyPath);
+            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(PackageName);
+            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(PackageVersion);
+            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(PackageSourceUrl);
+            hashCode = hashCode * -1521134295 + EqualityComparer<HashSet<string>>.Default.GetHashCode(Usings);
```

Comment:
Pretty sure that this is wrong, no?

Reply:
This was generated by VS, but I will use R# to do it.